### ISSUE TITLE
Caught deadly signal ABRT

### ISSUE THREAD

**Issue by hstarmans** [vim/vim#1550]
Dear all,

I really like vim! As a result, I am trying to compile vim with conda python 3.6. I have used the following [tutorial](https://github.com/Valloric/YouCompleteMe/wiki/Building-Vim-from-source.).
I use the following configuration:
```
./configure --with-features=normal \
            --enable-python3interp \
            --with-python3-config-dir=/home/myusername/anaconda3/lib/python3.6/config-3.6m-x86_64-linux-gnu \
            --prefix=/usr
```
Vim compiles without errors. 
```
vim --version 
```
The above command lists +python3 as supported.
Vim starts up, however if run a python3 command in vim, e.g.;
```
:python3 import os
```
Vim shuts down and I get the error "Caught deadly signal ABRT".
I have tried changing ~/vim/src/Makefile, creating a symbolic link file in /usr/local/bin and tried debugging VIM with gdb.
Gdb states ../sysdeps/unix/sysv/linux/raise.c:  No such file or directory.
The error is not very helpful, I am unable to get a more detailed error.
My OS is Ubuntu 16.04 Can I get a more detailed trace!? Where is vim looking for Python3!?

Any advice!?

**Comment by grochmal** [comment]
Conda mangles a lot of shared library references.  You will either need to find the conda environment and build Vim in that environment (i.e. as a conda package with all its sheanigans) or use a python installed without those mad shared library cross references.

There ain't a way for build systems (e.g. autoconf) to be aware of build environments (which are not OSes themselves) since build environments meddle into build systems.  That would be a circular dependency.

**Comment by chrisbra** [comment]
it is probably loading the wrong library. I am not sure, but it might be possible to set `'pythonthreedll'`

**Comment by hstarmans** [comment]
@chrisbra 
I hoped, i could inspect or change the variable via GDB. However, I can not find a global with the command 
```
info variables python* 
```
see gdb cheat sheet [GDB reference](https://www.scribd.com/document/2060261/GDB-Quick-Referencel).  
@grochmal 
I read some comments of people who succeeded in compiling vim with conda. This needs to be done for the vim plugin youcomplete me. There is a mention of conda in [issue 8](https://github.com/Valloric/YouCompleteMe/issues/8).
I am currently looking at a different solution where the Python environment can be specified with the [vim-conda plugin](https://github.com/cjrh/vim-conda).  Vim needs to be aware of different python environments for syntax checking and code suggestions. This would also solve my problems.

**Comment by dpelle** [comment]
@hstarmans  wrote:

> I read some comments of people who succeeded in compiling vim with conda.
> This needs to be done for the vim plugin youcomplete me. There is a mention
> of conda in issue 8.

I don't know what conda is, but I use the youcompleme plugin successfully,
along with rtag plugin with vim on ubuntu-16.04.  I built vim myself with:

$ ./configure --with-features=huge --enable-gui=gtk3 --enable-python3interp=yes

I used libpython from xubuntu packages.

**Comment by grochmal** [comment]
conda is basically a package manager whose philosophy is that python's pip is not enough to track python dependencies, since several python packages require compilation from C.  Conda compiles and re-packages these C based dependencies and places them in `/sysdeps/` (yes, it does not integrate with an existing package manager in any way).  Then mangles shared library paths to load python packages directly from there.

It works great if the top of your stack is python (i.e. something written in python).  But often fails miserably when you need to compile against the headers and the libraries of python itself, which is what Vim does.  Conda is an environment to use python, not to compile against python.

Basically, you're hammering a nail with a screwdriver.

Not impossible to do, you just need to find what shared libs from the OS (maintained by the OS' package manager and residing in `/usr/`) are missing from the subdirectory os `/sysdeps/` and compile them into there.  Which turns to be almost making a conda package for Vim.

I'd say that that's too painful to try because using a self compiled python or OS python should be much easier.  Yet, it is certainly possible to do (just don't ask me to do it).

**Comment by hstarmans** [comment]
Dear all,

Thanks for all the feedback. I guess the issue has been resolved. I am able to compile vim with the default ubuntu 16.04 python 3. 
@grochmal thanks for your in depth explanation why using conda is like hammering a nail with a screwdriver. 
I will try to get the syntax checking via the vim-conda plugin.
@dpelle ; I can recommend conda; especially if you build code which depends on libraries like VTK, Opencv etc. . Conda speeds up the installation of libraries.

**Comment by chrisbra** [comment]
I am closing it then.
